SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -467,6 +464,37 @@ void UGlobalStateManager::QueryGSM(const QueryDelegate& Callback)
 	Receiver->AddEntityQueryDelegate(RequestID, GSMQueryDelegate);
 }
 
+void UGlobalStateManager::QueryTranslation()
+{
+	// Build a constraint for the Virtual Worker Translation.
+	Worker_ComponentConstraint TranslationComponentConstraint{};
+	TranslationComponentConstraint.component_id = SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID;
+
+	Worker_Constraint TranslationConstraint{};
+	TranslationConstraint.constraint_type = WORKER_CONSTRAINT_TYPE_COMPONENT;
+	TranslationConstraint.constraint.component_constraint = TranslationComponentConstraint;
+
+	Worker_EntityQuery TranslationQuery{};
+	TranslationQuery.constraint = TranslationConstraint;
+	TranslationQuery.result_type = WORKER_RESULT_TYPE_SNAPSHOT;
+
+	Worker_RequestId RequestID;
+	RequestID = NetDriver->Connection->SendEntityQueryRequest(&TranslationQuery);
+
+	EntityQueryDelegate TranslationQueryDelegate;
+	TranslationQueryDelegate.BindLambda([this](const Worker_EntityQueryResponseOp& Op)
+	{
+		if (Op.status_code == WORKER_STATUS_CODE_SUCCESS)
```

Comment:
If the status code isn't success, this will be retried from the NetDriver.

Reply:
I think we should either explicitly retry here or set some bool flag to say there isn't something in flight that lets some other part of the codebase retry?

Reply:
Ah good point, I need to move the in flight reset.

Reply:
Nope, I do have it in the right place. Whether it's a success or failure, we'll clear the inFlight flag, so when NetDriver retries, it will work.